### ISSUE TITLE
Debugger Timeout when using Appium .NET

### ISSUE THREAD

**Issue by cosminstirbu** [mono/mono#8429]
## Steps to Reproduce

Check out https://github.com/cosminstirbu/appium-dotnet-driver/tree/bug/mono-debugger-timeout

1. Set a breakpoint in `iOSMonoDebugTimeoutTest` line 40 `(System.Diagnostics.Debug.Write(driver.PageSource);`
2. Debug `public void ScrollToExactTestCase()` test
3. Step over line 40, notice that the content is printed in the console
4. Add a watch on `driver.PageSource` - notice that the Debugger times out.

This reproduces on both Visual Studio for Mac and Rider. It DOESN'T reproduce on Visual Studio for PC.

## Expected Behavior

Debugger should not time out

### On which platforms did you notice this

[X] macOS
[ ] Linux
[ ] Windows

**Version Used**:

Mono JIT compiler version 5.8.1.0 (2017-10/6bf3922f3fd Thu Mar  8 17:00:31 EST 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          yes(3.6.0svn-mono-master/8b1520c8aae)
	GC:            sgen (concurrent by default)

**Comment by cosminstirbu** [comment]
Hello,

Any info on this?

Thank you,
Cosmin

**Comment by cosminstirbu** [comment]
Hello,

Any updates on this ticket?

Thanks,
Cosmin

**Comment by thaystg** [comment]
@cosminstirbu , I'm trying to reproduce this bug, but I think that I did something wrong in my installation. I run the appium doctor and it says that is everything working.
Then I start the appium server:
 appium
[Appium] Welcome to Appium v1.10.1
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
But when I try to run the unit test, I got this exception: 
The remote server returned an error: (500) Internal Server Error.

Coould you help me to reproduce this bug?
